finat/mixed.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import, print_function, division
+from six import iteritems
+from six.moves import zip
+
+import numpy
+
+import gem
+
+from finat.finiteelementbase import FiniteElementBase
+from finat.enriched import EnrichedElement
+
+
+def MixedElement(elements):
+    """Constructor function for FEniCS-style mixed elements.
+
+    Implements mixed element using :py:class:`EnrichedElement` and
+    value shape transformations with :py:class:`MixedSubElement`.
+    """
+    sizes = [numpy.prod(element.value_shape, dtype=int)
+             for element in elements]
+    offsets = [int(offset) for offset in numpy.cumsum([0] + sizes)]
+    total_size = offsets.pop()
+    return EnrichedElement([MixedSubElement(element, total_size, offset)
+                            for offset, element in zip(offsets, elements)])
```

Comment:
OK, so you're using the same technology to pick these apart as already exists, and the magic is how you make the tabulation of the sub elements.